README.md

```diff
@@ -0,0 +1,779 @@
+  />
+
+  // Passing different upload API options
+  <IKUpload
+    fileName="file-name.jpg"
+    tags={["sample-tag1", "sample-tag2"]}
+    customCoordinates={"10,10,10,10"}
+    isPrivateFile={false}
+    useUniqueFileName={true}
+    responseFields={["tags"]}
+    folder={"/sample-folder"}
+    ref={uploadRef}
+    onError={onError} onSuccess={onSuccess}
+    transformation={{
+      pre: 'l-text,i-Imagekit,fs-50,l-end', 
+      post: [
+        {
+          'type': 'transformation', 
+          'value': 'w-100'
+        }
+      ]
+    }}
+  />
+</ImageKitProvider>
+```
+
+## Demo application
+* The official step-by-step Next.js quick start guide - https://imagekit.io/docs/integration/next
+
+## Components
+
+The library includes 6 Components:
+
+* [`ImageKitProvider`](#ImageKitProvider) is used to define options such as `urlEndpoint`, `publicKey`, or `authenticator` for all child components. It does not render any UI elements.
+* `IKImage` for [image resizing](#image-resizing). This utilizes next/image and renders an `<img>` tag.
+* `IKVideo` for [video resizing](#video-resizing). This renders a `<video>` tag.
+* `IKUpload`for client-side [file uploading](#file-upload). This renders a `<input type="file">` tag.
+* `ImageKitClient` for [Core SDK](#imagekitclient), This exposes methods from [ImageKit javascript SDK](https://github.com/imagekit-developer/imagekit-javascript) like url and upload.
+* [`ImageKitContext`](#ImageKitContext) is a context used to provide access to options such as `urlEndpoint`, `publicKey`, `ikClient` or `authenticator` to child components within `ImageKitProvider`. It does not render any UI elements.
+
+## ImageKitProvider
+
+To use this SDK, you need to provide it with a few configuration parameters. You can use a parent `ImageKitProvider` component to define common options for all children `IKImage`, `IKVideo` or `IKupload` components. For example:
+
+```js
+<ImageKitProvider
+  urlEndpoint="https://ik.imagekit.io/your_imagekit_id"  // Required. Default URL-endpoint is https://ik.imagekit.io/your_imagekit_id
+  publicKey="your_public_api_key" // optional
+  transformationPosition="path" // optional
+  authenticator={()=>Promise} // optional
+  <IKImage path="/default-image.jpg" width={200} height={200} alt="Alt text"/>
+</ImageKitProvider>
+```
+
+* `urlEndpoint` is required to use the SDK. You can get URL-endpoint from your ImageKit dashboard - https://imagekit.io/dashboard/url-endpoints.
+* `publicKey` and `authenticator` parameters are required if you want to use the SDK for client-side file upload. You can get these parameters from the developer section in your ImageKit dashboard - https://imagekit.io/dashboard/developer/api-keys.
+* `transformationPosition` is optional. The default value for this parameter is `path`. Acceptable values are `path` & `query`
+
+> Note: Do not include your [private key](https://docs.imagekit.io/api-reference/api-introduction/api-keys#private-key) in any client-side code.
```

Comment:
Please replace all docs.imagekit.io links with valid new imagekit.io/docs link.

Reply:
Done updated the links